server/src/pages/shot/shotpage-header.js

```diff
@@ -0,0 +1,187 @@
+const { TimeDiff } = require("./time-diff");
+
+exports.ShotPageHeader = class ShotPageHeader extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  renderMyShotsText() {
+    // FIXME: this means that on someone else's shot they won't see a My Shots link:
+    if (!this.props.isOwner) {
+      return (
+        <Localized id="shotPageBackToHomeButton">
+        <button className="nav-button" title="Homepage" onClick={this.onClickMyShots.bind(this)}>
+          <div className="back-to-home">
+            <h2>
+              Firefox
+            </h2>
+            <h3>
+              Screenshots
+            </h3>
+          </div>
+        </button>
+      </Localized>
+      );
+    }
+    return (
+      <Localized id="shotPageAllShotsButton">
+        <button className="nav-button icon-shots" title="All Shots" onClick={this.onClickMyShots.bind(this)}>
+          <Localized id="shotPageAllShots">
+            <span>All Shots</span>
+          </Localized>
+        </button>
+      </Localized>
+    );
+  }
+
+  onClickMyShots() {
+    if (this.props.isOwner) {
+      sendEvent("goto-myshots", "navbar", {useBeacon: true});
+      location.href = "/shots";
+    } else {
+      sendEvent("goto-homepage", "navbar", {useBeacon: true});
+      location.href = "/";
+    }
+    // Note: we allow the default action to continue
+  }
+
+  renderShotInfo() {
+    const shot = this.props.shot;
+    const linkTextShort = shot.urlDisplay;
+    const timeDiff = <TimeDiff date={shot.createdDate} />;
+    return (
+      <div className="shot-main-actions">
+        <div className="shot-info">
+          <EditableTitle title={shot.title} isOwner={this.props.isOwner} />
+          <div className="shot-subtitle">
+            { linkTextShort ? <a className="subtitle-link" rel="noopener noreferrer" href={ shot.url } target="_blank" onClick={ this.onClickOrigUrl.bind(this, "navbar") }>{ linkTextShort }</a> : null }
+            <span className="time-diff">{ timeDiff }</span>
+          </div>
+        </div>
+      </div>
+    );
+  }
+
+  onClickOrigUrl(label) {
+    if (this.props.isOwner) {
+      sendEvent("view-original", `${label}-owner`, {useBeacon: true});
+    } else {
+      sendEvent("view-original", `${label}-non-owner`, {useBeacon: true});
+    }
+    // Note: we allow the default action to continue
+  }
+
+  renderFxASignIn() {
+    return (
+      this.props.isOwner ?
+        <div className="shot-fxa-signin">
+          <SignInButton isAuthenticated={this.props.isFxaAuthenticated} initialPage={this.props.shot.id} />
+        </div> : null
+    );
+  }
+
+  render() {
+    const myShotsText = this.renderMyShotsText();
+    const signin = this.renderFxASignIn();
+    const shotInfo = this.renderShotInfo();
+    return (
+      <Header>
+        { myShotsText }
+        { shotInfo }
+        <div className="shot-alt-actions">
+        { this.props.children }
+        { signin }
+        </div>
```

Comment:
Shot page specific header is rendering AllShots, Shot title and SignIn button. This has moved ~120 lines of code from Shot page view to ShotPageHeader. To keep this PR simple I have kept  Favorite, Download, Delete, Edit button with their new UI in view page. Ideally these should be moved in their own components and invoked from ShotPageHeader which can be done in subsequent iterations